bootstrap.go

```diff
@@ -42,15 +43,34 @@ func Start(ctx context.Context, opts ...Option) (StopFunc, error) {
 
 	l, err := datadogLogger.NewLogger(datadogLogger.WithGlobalLogger())
 	if err != nil {
-		return noop, fmt.Errorf("Failed to initialize the Datadog logger: %w", err)
+		return noop, fmt.Errorf("failed to initialize the Datadog logger: %w", err)
 	}
 	ddtrace.UseLogger(l)
 
+	// Make sure stop() only runs once
+	var once sync.Once
+	var cancelErr error
 	cancel := func() error {
-		return stop(options)
+		once.Do(func() {
+			cancelErr = stop(options)
+		})
+		return cancelErr
 	}
 
 	err = start(options)
+
+	if err != nil {
+		return noop, err
+	}
+
+	// Start cleaning up as soon as possible if the context is cancelled.
+	// Note: Any error from this cancel() call will also be returned on later invocations
+	// by any dependent program or system.
+	go func() {
+		<-ctx.Done()
+		_ = cancel() // nolint:errcheck // Error is intentionally ignored; it will be reported on subsequent cancel calls.
+	}()
```

Comment:
Have we considered the case of finishing traces etc as other code path is reacting to a canceled context?

What happens to a trace that surrounds a batch job that takes a while to complete when a signal to terminate the job is received? Would adding early flushing cause the trace not to be uploaded?

How would early flushing affect this trace: https://github.com/coopnorge/store-information-service/blob/53615a732fdf53a4072ffc10fd60541fc64c8ba3/store-information/cmd/db-sync/main.go#L64-L67